### ISSUE TITLE
[Question] Using subset of sequences

### ISSUE THREAD

**Issue by JoakimHaurum** [NVIDIA/nvvl#67]
Hi,

I have been looking through the code and was wondering whether it is possible to only use a subset of sequences obtained from a video. E.g. if we have a 100 frame video, and sequence length of 5, we get 96 valid sequences. When training is it then possible to utilize e.g. only 50% of these sequences? And if we have several videos, would it still be possible to shuffle the sequences?

**Comment by JoakimHaurum** [comment]
Hi Jared,

Thanks for the reply. I somehow totally overlooked the sample part of VideoLoader.

I made the following Sampler, which I supply the VideoLoader during initialization, in case other people need something similar

`class RandomSubsetVidSampler(torch.utils.data.sampler.Sampler):
    
    def __init__(self, data_source, share=0.5):
        self.data_source = data_source
        assert share > 0.0 and share <= 1.0, "Share input should be a value between 0 and 1"
        self.subset = int(len(data_source)*share)
        assert self.subset > 0, "The subset chosen is too small. Original Size: {}\nShare: {}\nSubset: {}".format(len(self.data_source), share, self.subset)

    def __iter__(self):
        return iter(np.random.choice(len(self.data_source),
                                     size=self.subset,
                                     replace=False).tolist())

    def __len__(self):
        return self.subset`
